### PR TITLE
Scan/ScanJob model/API/CLI refactoring

### PR DESCRIPTION
**Pull request by kholdaway**

### CODE REVIEW

File: docs/source/man.rst [hunk 1]
```diff
@@ -308,84 +308,145 @@ As the network infrastructure changes, it might be necessary to delete some sour
...
+Use the ``qpc scan`` command to create, run and manage scans.
 
-**IMPORTANT:** If any ssh-agent connection is set up for a target host, that connection will be used as a fallback connection.
+A scan defines a collection of network information, including the sources to scan and additional options, such as elective products to omit from the scan and the maximum number of parallel system scans. The creation of a scan references information such as sources so that the act of running the scan is repeatable, without a requirement to reenter network information for each scan attempt.
+
+Creating and Editing Scans
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Use the ``qpc scan add`` command to create scan objects with one or more sources. This command creates a scan object that references the supplied sources and contains any options supplied by the user. Each instance of a scan object is assigned a unique *identifier* to identify the scan jobs that are associated with that scan.
```
Comment (chambridge) [inline docs/source/man.rst]: I think we can drop this portion of the doc:

> Each instance of a scan object is assigned a unique *identifier* to identify the scan jobs that are associated with that scan.

File: docs/source/man.rst [hunk 2]
```diff
@@ -308,84 +308,145 @@ As the network infrastructure changes, it might be necessary to delete some sour
...
 
-**qpc scan start --sources=** *source_list* **[--max-concurrency=** *concurrency* **]** **--disable-optional-products=** *products_list*
+**qpc scan add --name** *name* --sources=** *source_list* **[--max-concurrency=** *concurrency* **]** **--disable-optional-products=** *products_list*
```
Comment (chambridge) [inline docs/source/man.rst]: The disable-optional-producst should have been in brackets:

```
**[--disable-optional-products=** *products_list* **]**
```

File: docs/source/man.rst [hunk 3]
```diff
@@ -308,84 +308,145 @@ As the network infrastructure changes, it might be necessary to delete some sour
...
+  Optional. The product inspection exclusion. Contains the list of products to exclude from inspection. Valid values are jboss_eap, jboss_fuse, and jboss_brms.
+
+The information in a scan might change as the structure of the network changes. Use the ``qpc scan edit`` command to edit an existing scan to accommodate those changes.
+
+Although ``qpc scan`` options can accept more than one value, the ``qpc scan edit`` command is not additive. To edit a scan and add a new value for an option, you must enter both the current and the new values for that option. Include only the options that you want to change in the ``qpc scan edit`` command. Options that are not included are not changed.
+
+**qpc scan edit --name** *name* --name** *name* --sources=** *source_list* **[--max-concurrency=** *concurrency* **]** **--disable-optional-products=** *products_list*
+
+For example, if a scan contains a value of ``network1sources`` for the ``--sources`` option, and you want to change that scan to use both the ``network1sources`` and ``network2sources`` sources, you would edit the scan as follows:
```
Comment (chambridge) [inline docs/source/man.rst]: would be better with singular names:
`network1source` and `network2source`

Comment (chambridge) [reply docs/source/man.rst]: or better `satellite1source` to show you can mix source types.

File: docs/source/man.rst [hunk 4]
```diff
@@ -308,84 +308,145 @@ As the network infrastructure changes, it might be necessary to delete some sour
...
 
 ``--type=type``
 
   Optional. Filters the results by scan type. This value must be ``connect`` or ``inspect``. A scan of type ``connect`` is a scan that began the process of connecting to the defined systems in the sources, but did not transition into inspecting the contents of those systems. A scan of type ``inspect`` is a scan that moves into the inspection process.
 
-``--state=state``
+The ``qpc scan job`` command returns the the list of scan jobs or a single scan job associated with a scan object. The output of this command includes the scan job identifiers as well as the status of each job and the results.
 
-  Optional. Filters the results by scan state. This value must be ``created``, ``pending``, ``running``, ``paused``, ``canceled``, ``completed``, or ``failed``.
+**qpc scan job** --name** *scan_name* **--id=** *scan_job_identifier* **--status=** *(created | pending | running | paused | canceled | completed | failed)*
```
Comment (chambridge) [inline docs/source/man.rst]: Let's open a separate issue to remove the requirement on scan_name since we don't strictly need this for the API request.

Comment (chambridge) [reply docs/source/man.rst]: We should move this to the scanning section below.

File: docs/source/man.rst [hunk 5]
```diff
@@ -308,84 +308,145 @@ As the network infrastructure changes, it might be necessary to delete some sour
...
+
+**qpc scan show --name** *name*
+
+``--name=name``
+
+  Required. Contains the name of the scan object to display.
+
+The ``qpc scan status`` command returns the status details for a single specified scan job.
+
+**qpc scan status** **--id=** *scan_job_identifier*
```
Comment (chambridge) [inline docs/source/man.rst]: We should move this to the scanning section below.

File: docs/source/man.rst (hunk 1 above)
Comment (kholdaway) [reply docs/source/man.rst]: Removed

File: docs/source/man.rst (hunk 2 above)
Comment (kholdaway) [reply docs/source/man.rst]: Done

File: docs/source/man.rst (hunk 3 above)
Comment (kholdaway) [reply docs/source/man.rst]: Done

File: docs/source/man.rst (hunk 5 above)
Comment (kholdaway) [reply docs/source/man.rst]: Done.

**Pull request merged**